tests/e2e/utils/src/components.js

```diff
@@ -450,23 +454,29 @@ const addProductToOrder = async ( orderId, productName ) => {
  * @param discountType Type of a coupon. Defaults to Fixed cart discount.
  */
 const createCoupon = async ( couponAmount = '5', discountType = 'Fixed cart discount' ) => {
-	await merchant.openNewCoupon();
+	let couponType;
+	switch ( discountType ) {
```

Comment:
Not that we need to change this here, but this made me wonder if it's easier to pass in the API name (e.g. `fixed_cart`, `fixed_product` etc) compared with the UI name, which could change.

Reply:
> Not that we need to change this here, but this made me wonder if it's easier to pass in the API name

We could make that the `default:` which would allow updating the parameter but keep it backward compatible.

> It seems like these tests are failing pretty consistently for me 

Did you run `npm run build:packages` on this branch?

Reply:
> Did you run npm run build:packages on this branch?

I did when I initially tested (also just did another run-through to check for sure) and I am still seeing those failures, unfortunately.

Reply:
Just to follow up on the above, it seems it was some flakiness I was hitting locally. After making the changes on this PR https://github.com/woocommerce/woocommerce/pull/29743 I no longer was hitting the repeated failures above.

Reply:
> Not that we need to change this here, but this made me wonder if it's easier to pass in the API name (e.g. fixed_cart, fixed_product etc) compared with the UI name, which could change, especially with the changes to use the API below.

@zhongruige I updated to include support for this. This is ready for another review.